client/packages/inventory/src/Stocktake/DetailView/modal/StocktakeLineEdit/hooks/useStocktakeLineEdit.ts

```diff
@@ -1,31 +1,40 @@
-import { RecordPatch, ArrayUtils } from '@openmsupply-client/common';
-import { ItemRowFragment } from '@openmsupply-client/system';
 import {
-  StocktakeLineFragment,
-  UpsertStocktakeLinesMutation,
-  useStocktake,
-} from './../../../../api';
+  RecordPatch,
+  ArrayUtils,
+  useTranslation,
+  noOtherVariants,
+} from '@openmsupply-client/common';
+import { ItemRowFragment } from '@openmsupply-client/system';
+import { StocktakeLineFragment, useStocktake } from './../../../../api';
 import { DraftStocktakeLine, DraftLine } from '../utils';
 import { useNextItem } from './useNextItem';
 import { useDraftStocktakeLines } from './useDraftStocktakeLines';
+import { useStocktakeLineErrorContext } from 'packages/inventory/src/Stocktake/context';
 
 interface useStocktakeLineEditController {
   draftLines: DraftStocktakeLine[];
   update: (patch: RecordPatch<StocktakeLineFragment>) => void;
   addLine: () => void;
-  save: (lines: DraftStocktakeLine[]) => Promise<UpsertStocktakeLinesMutation>;
+  save: () => Promise<{ errorMessages?: string[] }>;
   isLoading: boolean;
   nextItem: ItemRowFragment | null;
-  isError: boolean;
 }
 
+export const errorMessage = (error: unknown): string => {
+  // Bugsnag it ?
+  if (error instanceof Error) return error.message;
+  return String(error);
+};
+
 export const useStocktakeLineEdit = (
   item: ItemRowFragment | null
 ): useStocktakeLineEditController => {
+  const t = useTranslation(['inventory']);
   const { id } = useStocktake.document.fields('id');
   const nextItem = useNextItem(item?.id);
   const [draftLines, setDraftLines] = useDraftStocktakeLines(item);
-  const { mutateAsync: save, isLoading, isError } = useStocktake.line.save();
+  const { mutateAsync, isLoading } = useStocktake.line.save();
```

Comment:
side note: I still prefer the renaming of `mutateAsync` - it's such a vague fn name 🙄

Reply:
🙏 it was renamed here: e58e8dc9cf3f2b8e680a0954fd6e90165ae51729